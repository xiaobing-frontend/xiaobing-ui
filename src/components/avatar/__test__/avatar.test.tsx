import React from 'react'
import { render } from '@testing-library/react'

import Avatar, { AvatarProps } from '../avatar'

const url =
  'https://ss3.bdstatic.com/70cFv8Sh_Q1YnxGkpoWK1HF6hhy/it/u=1068316074,4162025599&fm=26&gp=0.jpg'

const testProps: AvatarProps = {
  src: url,
  size: 'large',
  shape: 'circle',
}
describe('test Avatar component', () => {
  it('should render correct component', () => {
    const wrapper = render(<Avatar src={url} shape="circle" />)
    const element = wrapper.container.querySelector('.art-avatar') as HTMLElement
    expect(element.tagName).toEqual('SPAN')
    expect(element).toHaveClass('art-avatar art-avatar-default')
    expect(wrapper).toMatchSnapshot()
  })
  it('should render correct size', () => {
    const wrapper = render(<Avatar {...testProps} />)
    const element = wrapper.container.querySelector('.art-avatar') as HTMLElement
    expect(element.tagName).toEqual('SPAN')
    expect(element).not.toHaveClass('art-avatar-default')
    expect(element).toHaveClass('art-avatar-large')
  })
})
