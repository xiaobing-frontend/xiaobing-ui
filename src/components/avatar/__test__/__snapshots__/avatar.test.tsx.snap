// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Avatar component sholud render correct component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span
        class="art-spin art-spin-dot-spin art-spin-spinning"
      >
        <i
          class="art-spin-dot-item"
        />
        <i
          class="art-spin-dot-item"
        />
        <i
          class="art-spin-dot-item"
        />
        <i
          class="art-spin-dot-item"
        />
      </span>
    </div>
  </body>,
  "container": <div>
    <span
      class="art-spin art-spin-dot-spin art-spin-spinning"
    >
      <i
        class="art-spin-dot-item"
      />
      <i
        class="art-spin-dot-item"
      />
      <i
        class="art-spin-dot-item"
      />
      <i
        class="art-spin-dot-item"
      />
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`test Avatar component should render correct component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span
        class="art-avatar art-avatar-default art-avatar-circle"
      >
        <img
          src="https://ss3.bdstatic.com/70cFv8Sh_Q1YnxGkpoWK1HF6hhy/it/u=1068316074,4162025599&fm=26&gp=0.jpg"
        />
      </span>
    </div>
  </body>,
  "container": <div>
    <span
      class="art-avatar art-avatar-default art-avatar-circle"
    >
      <img
        src="https://ss3.bdstatic.com/70cFv8Sh_Q1YnxGkpoWK1HF6hhy/it/u=1068316074,4162025599&fm=26&gp=0.jpg"
      />
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
